File: src/main/java/com/vaadin/demo/component/datepicker/DatePickerAutoOpen.java
package com.vaadin.demo.component.datepicker;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.datepicker.DatePicker;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

@Route("date-picker-auto-open")
public class DatePickerAutoOpen extends Div {

    public DatePickerAutoOpen() {
        DatePicker datePicker = new DatePicker("Start date");
        // tag::snippet[]
        datePicker.setAutoOpen(false);
        // end::snippet[]

        add(datePicker);
    }

    public static class Exporter extends DemoExporter<DatePickerAutoOpen> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/datepicker/date-picker-auto-open.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/date-picker';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('date-picker-auto-open')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-date-picker label="Start date" auto-open-disabled></vaadin-date-picker>
      <!-- end::snippet[] -->
    `;
  }
}
